### PR TITLE
auction: define and record events

### LINKED ISSUE #4303: auction: define and emit events
We want to define three types of events
- [ ] `EventDutchAuctionScheduled`
- [ ] `EventDutchAuctionUpdated`
- [ ] `EventDutchAuctionEnded`

Each tracking sufficient data so that an explorer can cross-ref auction activity with position management happening on the dex.

### PR DESCRIPTION
**Pull request by erwanor**
## Describe your changes

This PR adds three messages to the `v1alpha1` auction interface:
- `EventDutchAuctionScheduled`
- `EventDutchAuctionUpdated`
- `EventDutchAuctionEnded`

## Issue ticket number and link

Fixes #4303.

## Checklist before requesting a review

- [x] If this code contains consensus-breaking changes, I have added the "consensus-breaking" label. Otherwise, I declare my belief that there are not consensus-breaking changes, for the following reason:

  > this only makes additive changes, recording events in the auction component.

### CODE REVIEW

**Comment by cratelyn** [comment]
i'll mark this as "ready for review" since reviews were requested. feel free to move it back to a draft if there's still work to be done here!

File: proto/penumbra/penumbra/core/component/auction/v1alpha1/auction.proto [hunk 1]
```diff
@@ -173,3 +173,30 @@ message ActionDutchAuctionWithdrawView {
...
+
+message EventDutchAuctionUpdated {
+  AuctionId auction_id = 1;
+  DutchAuctionState state = 2;
+}
+
+message EventDutchAuctionEnded {
+  // The reason the auction ended.
+  enum Reason {
+    REASON_UNSPECIFIED = 0;
```
Comment (cratelyn) [inline proto/penumbra/penumbra/core/component/auction/v1alpha1/auction.proto:190]: i see we never create this variant in `crates/core/component/auction/src/event.rs`. should we remove it? we could always add new variants in the future if needed.

**Review by cratelyn — APPROVED** [review]
this looks great!

File: proto/penumbra/penumbra/core/component/auction/v1alpha1/auction.proto (hunk 1 above)
Comment (cratelyn) [reply proto/penumbra/penumbra/core/component/auction/v1alpha1/auction.proto:190]: answered this question myself, via `rg 'enum \w* {' -C 10 proto/penumbra`.

i see this is an existing pattern, so i'll mark this as resolved.

Comment (hdevalence) [reply proto/penumbra/penumbra/core/component/auction/v1alpha1/auction.proto:190]: https://buf.build/docs/lint/rules#enum_value_prefix

All enums should have a zero value. proto3 doesn't differentiate between set and unset fields, so if an enum field isn't explicitly set, it defaults to the zero value. If an explicit zero value isn't part of the enum definition, this defaults to the actual zero value of the enum. For example, if you had the following .proto file, any Uri with scheme not explicitly set would default to SCHEME_FTP:

**Comment by cratelyn** [comment]
@erwanor **NB** i edited the description to use `Fixes #4303` so that this will automatically close that issue as completed. this appears to address all of the events outlined in that issue.

File: proto/penumbra/penumbra/core/component/auction/v1alpha1/auction.proto (hunk 1 above)
Comment (cratelyn) [reply proto/penumbra/penumbra/core/component/auction/v1alpha1/auction.proto:190]: thanks for the background @hdevalence & @erwanor! i appreciate hearing the underlying reason for these `FOO_UNSPECIFIED` variants.

**Comment by TalDerei** [comment]
LGTM!

**Pull request merged**